ncm-network/src/main/perl/network.pm

```diff
@@ -809,20 +933,49 @@ sub Configure
     } else {
         foreach my $if (sort keys %ifdown) {
             # how do we actually know that the device was up?
-            # eg for non-existing device eth4: /sbin/ifdown eth4 --> usage: ifdown <device name>
+            # Do not try to down non-existing devices
+            next if (-d "/sys/class/net" && ! -e "/sys/class/net/$if");
             push(@cmds, ["/sbin/ifdown", $if]);
         }
     }
     $self->runrun(@cmds);
     # replace modified/new files
+    my $was_change = 0;
     foreach my $file (keys %exifiles) {
         if (($exifiles{$file} == 1) || ($exifiles{$file} == 2)) {
             copy($self->gen_backup_filename($file).FAILED_SUFFIX,$file) ||
                 $self->error("replace modified/new files: can't copy ",
                              $self->gen_backup_filename($file).FAILED_SUFFIX,
                              " to $file. ($!)");
+            $was_change = 1;
         } elsif ($exifiles{$file} == -1) {
             unlink($file) || $self->error("replace modified/new files: can't unlink $file. ($!)");
+            $was_change = 1;
+        }
+    }
+    ## kick udev to rename devices if needed; do nothing if no files were changed
```

Comment:
see also #1012 
but we had some udev related issues with storage devices too, so maybe a CAF::udev might make sense

Reply:
see #1118 for storage udev PR